src/main/java/org/ow2/proactive/procci/service/occi/InstanceService.java

```diff
@@ -60,54 +57,48 @@
     @Autowired
     private CloudAutomationInstanceClient cloudAutomationInstanceClient;
 
-    @Autowired
-    private MixinService mixinService;
-
-    @Autowired
-    private TransformerManager transformerManager;
-
     /**
-     * Get a resource from the data stored in Cloud-automation-service
+     * Get an entity from the data stored in Cloud-automation-service
      *
      * @param id is the id of the entity
+     * @param transformerProvider the transformer provider for an entity inherited type
      * @return an entity
      * @throws ClientException
      */
-    public Optional<Entity> getEntity(String id) throws ClientException {
-        return cloudAutomationInstanceClient.getInstanceByVariable(ID_NAME, ConvertUtils.formatURL(id))
-                                            .map(model -> getResourceBuilder(model).addMixins(pullMixinFromCloudAutomation(id))
-                                                                                   .build());
+    public Optional<Entity> getEntity(String id, TransformerProvider transformerProvider) {
+        return cloudAutomationInstanceClient.getInstanceModel(ID_NAME, ConvertUtils.formatURL(id), transformerProvider)
+                                            .filter(instanceModel -> transformerProvider.isInstanceOfType(instanceModel))
```

Comment:
Again, why not delegating the work to the server? retrieving all required information on the client looks not scalable..